web/src/ui_init.js

```diff
@@ -483,6 +483,30 @@ function initialize_unread_ui() {
     unread_ui.initialize({notify_server_messages_read: unread_ops.notify_server_messages_read});
 }
 
+function initialize_topic_zoom() {
+    // Register post zoom-in hooks here.
+    topic_zoom.register_post_zoom_in_hook(() => popovers.hide_all_except_sidebars());
+    topic_zoom.register_post_zoom_in_hook(() => pm_list.close());
+    topic_zoom.register_post_zoom_in_hook(({stream_id}) =>
+        stream_list.zoom_in_topics({
+            stream_id,
+        }),
+    );
+
+    // Register post zoom-out hooks here.
+    topic_zoom.register_post_zoom_out_hook(
+        ({pending_stream_list_rerender}) =>
+            pending_stream_list_rerender ?? stream_list.update_streams_sidebar(true),
+    );
+    topic_zoom.register_post_zoom_out_hook(() => popovers.hide_all_except_sidebars());
+    topic_zoom.register_post_zoom_out_hook(() => stream_list.zoom_out_topics());
+    topic_zoom.register_post_zoom_out_hook(
+        ({stream_id}) => stream_id ?? stream_list.scroll_stream_into_view(stream_id),
+    );
+
+    topic_zoom.initialize();
+}
```

Comment:
Let's leave this commit off for now; it's a lot of messy complexity, and I'm skeptical that adding hooks here is the right approach. The `stream_list`, `topic_list`, and `topic_zoom` modules are entangled in a way that I expect will better be addressed by moving code, merging modules. Save the commit in case we want it, but can come back to this one when we've settled more of the other import cycles.

I think the others will end up being pretty non-controversial with my recommendations in the rest of this PR, so I'd like to avoid them getting slowed down by this change which might not merge.

Reply:
Okay sure dropped this commit!